### PR TITLE
CloudWatch: remove core imports from CloudWatchMetricsQueryRunner

### PR DESCRIPTION
**Pull request by kevinwcyu**
**What is this feature?**

Removes app notification related core imports from the CloudWatch `CloudWatchMetricsQueryRunner`

This PR partially removes the following core imports, remaining imports will be removed in separate PRs:
```ts
import { notifyApp } from 'app/core/actions';
import { createErrorNotification } from 'app/core/copy/appNotification';
import { store } from 'app/store/store';
import { AppNotificationTimeout } from 'app/types'
```

**Why do we need this feature?**

This work is part of the work towards decoupling CloudWatch from core Grafana

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

This change only affects throttling errors. They are currently displayed using a custom component, `ThrottlingErrorMessage`, that provides links to additional information about AWS quotas. The only difference is the error message is now enriched with some additional information that points the users to where they can learn more about the throttling and how to manage their CW quotas to reduce throttling errors.

Testing: instead of triggering an actual throttling exception, you can create an invalid query (e.g. `!`) in the code editor to trigger an exception and then modify the message in `public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchMetricsQueryRunner.ts` in the `performTimeSeriesQuery` function before the data is returned:

```ts
        if (res.errors?.length) {
          res.errors[0].message = `Throttling: exception`;
        }

         // add the above before returning the data
        return {
          data: dataframes,
          // DataSourceWithBackend will not throw an error, instead it will return "errors" field along with the response
          errors: this.enrichThrottlingErrorMessages(request, res.errors),
        };
```

**The error alert is still displayed at the top right corner, and the error message below the query editor provides an enriched error message**

https://github.com/grafana/grafana/assets/19530599/3a29e739-6f20-4f55-afbc-3ee4de11efac

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by idastambuk — APPROVED** [review]
Last time I worked on this I was unsure whether we should distinguish throttling errors from normal ones, so I let them be. But I support treating them as regular errors from now on. LGTM!

File: public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchMetricsQueryRunner.ts [hunk 1]
```diff
@@ -14,38 +13,18 @@ import {
...
-      createErrorNotification(
-        `CloudWatch request limit reached in ${region} for data source ${datasourceName}`,
-        '',
-        undefined,
-        React.createElement(ThrottlingErrorMessage, { region }, null)
-      )
-    )
-  );
+const getThrottlingErrorMessage = (region: string, message: string) =>
+  `Please visit the AWS Service Quotas console at https://${region}.console.aws.amazon.com/servicequotas/home?region=${region}#!/services/monitoring/quotas/L-5E141212 to request a quota increase or see our documentation at https://grafana.com/docs/grafana/latest/datasources/cloudwatch/#manage-service-quotas to learn more. ${message}`;
```
Comment (iwysiu) [inline public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchMetricsQueryRunner.ts:24]: Is it possible for these to be links, or is that just not how errors work?

**Review by iwysiu — APPROVED** [review]

File: public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchMetricsQueryRunner.ts (hunk 1 above)
Comment (kevinwcyu) [reply public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchMetricsQueryRunner.ts:24]: The `DataQueryError` interface only accepts strings for messages.

**Comment by kevinwcyu** [comment]
@idastambuk, @iwysiu, I've made some updates to this PR and would love a second look at it.

The main change is that I swapped the core imports to use the appEvents export from `@grafana/runtime` to publish the notification instead (this let's us keep the custom error component `ThrottlingErrorMessage` that provides links to manage AWS quotas).

**Review by iwysiu — COMMENTED** [review]
Nice! Having both sounds good. I pulled the changes locally though, and the custom error component didn't appear for me?

**Review by iwysiu — APPROVED** [review]
I tried it again to repro it and the alert popped up, I think my local repo just got itself into a weird state.

**Pull request merged**